src/test/kotlin/no/nav/familie/ef/sak/iverksett/oppgaveforbarn/BarnFyllerÅrOppfølgingsoppgaveServiceTest.kt

```diff
@@ -0,0 +1,176 @@
+
+        opprettOppgaveForBarnService.opprettOppgaverForAlleBarnSomHarFyltÅr()
+        verify(exactly = 5) { oppgaveRepository.insert(any()) }
+    }
+
+    @Test
+    fun `to barn som fyller år på forskjellige behandlinger, forvent at to oppgaver er gjeldende`() {
+        val fødselsdato = LocalDate.now().minusYears(1).minusDays(6)
+
+        val opprettBarnForFødselsdato = listOf(
+            opprettBarn(behandlingId = UUID.randomUUID(), fødselsnummer = FnrGenerator.generer(fødselsdato)),
+            opprettBarn(behandlingId = UUID.randomUUID(), fødselsnummer = FnrGenerator.generer(fødselsdato))
+        )
+        every {
+            gjeldendeBarnRepository.finnBarnAvGjeldendeIverksatteBehandlinger(StønadType.OVERGANGSSTØNAD, any())
+        } returns opprettBarnForFødselsdato
+
+        val opprettBarnTilOppgave = opprettBarnForFødselsdato.map { BarnTilOppgave(it.fødselsnummerBarn!!, it.behandlingId, 1, 1) }.toSet()
+        every { gjeldendeBarnRepository.finnEksternFagsakIdForBehandlingId(any()) } returns opprettBarnTilOppgave
+
+        opprettOppgaveForBarnService.opprettOppgaverForAlleBarnSomHarFyltÅr()
+        verify(exactly = 2) { oppgaveRepository.insert(any()) }
+    }
+
+    @Test
+    fun `barn fra vanlige behandlinger og migrerte fagsaker blir med i listen over oppgaver`() {
+
+        val fødselsdato = LocalDate.now().minusDays(182).minusDays(4)
+
+        val behandlingId = UUID.randomUUID()
+        val migrertBehandlingId = UUID.randomUUID()
+        val opprettetBarn = opprettBarn(behandlingId = behandlingId, FnrGenerator.generer(fødselsdato))
+        val opprettBarnMigrering = opprettBarn(
+            behandlingId = migrertBehandlingId,
+            fødselsnummer = FnrGenerator.generer(fødselsdato),
+            fraMigrering = true
+        )
+        val listOpprettedeBarn = listOf(opprettetBarn, opprettBarnMigrering)
+        every {
+            gjeldendeBarnRepository.finnBarnAvGjeldendeIverksatteBehandlinger(StønadType.OVERGANGSSTØNAD, any())
+        } returns listOf(opprettetBarn)
+
+        every {
+            gjeldendeBarnRepository.finnBarnTilMigrerteBehandlinger(StønadType.OVERGANGSSTØNAD, any())
+        } returns listOf(opprettBarnMigrering)
+
+        val opprettBarnTilOppgave = listOpprettedeBarn.map { BarnTilOppgave(it.fødselsnummerBarn!!, it.behandlingId, 1, 1) }.toSet()
+        every { gjeldendeBarnRepository.finnEksternFagsakIdForBehandlingId(any()) } returns opprettBarnTilOppgave
+
+        opprettOppgaveForBarnService.opprettOppgaverForAlleBarnSomHarFyltÅr()
+        verify(exactly = 2) { oppgaveRepository.insert(any()) }
+    }
+
+    private fun opprettBarn(
+        behandlingId: UUID = UUID.randomUUID(),
+        fødselsnummer: String? = null,
+        fraMigrering: Boolean = false
+    ): BarnTilUtplukkForOppgave {
+        return BarnTilUtplukkForOppgave(behandlingId, "12345678910", fødselsnummer, fraMigrering)
```

Comment:
Det var jo noen caser som gikk "under radaren" på den eksisterende løsningen vår - har du laget tester for disse? (Jeg husker jo ikke i farta hvilke caser det var :D )

Reply:
Typisk case som ble identifisert var de tilfellene der en behandling har blitt ferdigstilt rett før et barn har fylt 6 mnd. Disse ble ikke plukket opp fordi det ble gjort klart oppgaver frem i tid, basert på dagens data. Testene ser nå tilbake i tid, og er en slags case på det. La i tillegg til en test som viser at barn ikke får oppgave opprettet dersom behandlingen ikke er iverksatt.